src/thunderbots/software/network_input/filter/ball_filter.h

```diff
@@ -41,9 +48,21 @@ class BallFilter
      *
      * @return The updated state of the ball given the new data
      */
-    Ball getFilteredData(const Ball& current_ball_state,
-                         const std::vector<SSLBallDetection>& new_ball_detections);
+    std::optional<Ball> getFilteredData(const std::vector<SSLBallDetection>& new_ball_detections, const Field& field);
+
+//   private:
+    // TODO: comments
+    void addNewDetectionsToBuffer(std::vector<SSLBallDetection> new_ball_detections,
+                                  const Field &field);
+    Ball getAveragedBallPositionAndVelocity(boost::circular_buffer<SSLBallDetection> ball_detections);
+    Ball getLinearRegressionPositionAndVelocity(boost::circular_buffer<SSLBallDetection> ball_detections);
+    std::vector<Vector> calculateBallVelocities(
+            boost::circular_buffer<SSLBallDetection> ball_detections);
+
+
 
-   private:
-    boost::circular_buffer<Vector> previous_ball_readings;
+    // A circular buffer used to store previous ball detections, so we can use them
+    // in the filter
+    boost::circular_buffer<SSLBallDetection> ball_detection_buffer;
+    unsigned int LINEAR_REGRESSION_BUFFER_SIZE;
```

Comment:
nit; suggest `size_t` instead of `unsigned int`

Reply:
fixed in new version